docs_src/microservices/application/BuiltIn.md

```diff
@@ -0,0 +1,256 @@
+
+!!! example
+    ``` go
+    NewFilter([] {"Device1", "Device2"}).FilterByDeviceName
+    ```
+
+
+## Filtering
+
+There are two basic types of filtering included in the SDK to add to your pipeline. There is also an option to `Filter Out` specific items. Theses provided Filter functions return a type of events.Model. If filtering results in no remaining data, the pipeline execution for that pass is terminated. If no values are provided for filtering, then data flows through unfiltered.
+
+| Factory Method                   | Description | 
+|----------------------------------|-------------|
+| NewFilter([]string filterValues)  | This function returns a `Filter` instance initialized with the passed in filter values. This `Filter` instance is used to access the following filter functions that will operate using the specified filter values. |
+
+
+``` go
+type Filter struct {
+    // Holds the values to be filtered
+    FilterValues []string
+    // Determines if items in FilterValues should be filtered out. If set to true all items found in the filter will be removed. If set to false all items found in the filter will be returned. If FilterValues is empty then all items will be returned.
+	FilterOut    bool
+}
+```
+
+### By Device Name
+`FilterByDeviceName` - This function will filter the event data down to the specified device names and return the filtered data to the pipeline.
+
+``` go
+NewFilter([] {"Device1", "Device2"}).FilterByDeviceName
+```
+
+
+### By Value Descriptor
+`FilterByValueDescriptor` - This function will filter the event data down to the specified device value descriptor and return the filtered data to the pipeline.
+
+``` go
+NewFilter([] {"ValueDescriptor1", "ValueDescriptor2"}).FilterByValueDescriptor
+```
+
+
+### JSON Logic
+| Factory Method                   | Description | 
+|----------------------------------|-------------|
+| NewJSONLogic(rule string) | This function returns a `JSONLogic` instance initialized with the passed in JSON rule. The rule passed in should be a JSON string conforming to the specification here: http://jsonlogic.com/operations.html. |
+
+`Evaluate` - This is the function that will be used in the pipeline to apply the JSON rule to data coming in on the pipeline. If the condition of your rule is met, then the pipeline will continue and the data will continue to flow to the next function in the pipeline. If the condition of your rule is NOT met, then pipeline execution stops. 
+
+
+``` go
+NewJSONLogic("{ \"in\" : [{ \"var\" : \"device\" }, [\"Random-Integer-Device\",\"Random-Float-Device\"] ] }").Evaluate
+```
+
+!!! note
+    Only  operators that return true or false are supported. See http://jsonlogic.com/operations.html# for the complete list paying attention to return values. Any operator that returns manipulated data is currently not supported. For more advanced scenarios checkout [EMQ X Kuiper](https://github.com/emqx/kuiper).
```

Comment:
Any operator that returns manipulated data is currently not supported...  
Not sure what this means.  Do you mean only comparison operators are supported?

Reply:
I changed operators -> operations. It is what JSON logic call them on their site if you click the link